### PR TITLE
uplift esaml to otp22/cowboy 2.7

### PR DESCRIPTION
**Pull request by anderswei**

### CODE REVIEW

File: src/esaml_cowboy.erl [hunk 1]
```diff
@@ -93,29 +91,22 @@ reply_with_req(IDP, SignedXml, Additional_Query_Params, RelayState, Req) ->
...
         <<"POST">> ->
-            % XXX: compat hack, the cowboy_req:continue/1 function was introduced at the
-            %      same time as the API change on body_qs/2, so we can use it to detect
-            %      which argument layout we need to use. this way we can be compat with
-            %      both cowboy <1.0 and 1.0.x.
-            {ok, PostVals, Req2} = case erlang:function_exported(cowboy_req, continue, 1) of
-                true -> cowboy_req:body_qs(Req, [{length, 128000}]);
-                false -> cowboy_req:body_qs(128000, Req)
-            end,
+            {ok, PostVals, Req2} = cowboy_req:read_urlencoded_body(Req, #{length => 120000}),
```
Comment (TC-vLu) [inline src/esaml_cowboy.erl]: it should be 128000, isn't it?

**Review by TC-aLi — COMMENTED** [review]
I feel we can move every thing in this repo to saml and get rid of this repo totally. Upstream is not being maintained anymore.

File: rebar.config [hunk 2]
```diff
@@ -1,7 +1,8 @@
+{erl_opts,[warnings_as_errors]}.
 {deps, [
-    {erlang_commons,    "1.1.16", {git, "git@github.com:tigertext/erlang_commons.git",  {tag, "1.1.16"}}},
```
Comment (zvoykish) [inline rebar.config:2]: Was that simply unused?

File: src/esaml_cowboy.erl [hunk 3]
```diff
@@ -210,6 +198,14 @@ perform_extra_security_if_applicable(Callback,   Callback_State,  Xml, Assertion
         {error, E}  -> {error, E, Req}
     end.
 
+qs_val(Key, Req) when is_atom(Key) ->
```
Comment (zvoykish) [inline src/esaml_cowboy.erl:201]: I'd just remove the Req from the response altogether instead of passing it back & forth

**Review by zvoykish — APPROVED** [review]
While I'm not familiar enough with the otp22 & cowboy 2.x syntax - this looks good from reviewing the changes to the esaml library

File: rebar.config (hunk 2 above)
Comment (anderswei) [reply rebar.config:2]: yes it is not used.

**Pull request merged**